models/FollowModel.js

```diff
@@ -17,8 +17,37 @@ const getFollowStatus = `
   ) as "followStatus";
 `;
 
+const getFollowersUserList = `
+  SELECT 
+    u."userId", 
+    u."username", 
+    u."displayName", 
+    CASE
+      WHEN f2."followerUserId" IS NOT NULL THEN TRUE
+      ELSE FALSE
+    END AS "isFollowing"
+  FROM app_user u
+  INNER JOIN follow f1 ON u."userId" = f1."followerUserId"
+  LEFT JOIN follow f2 ON u."userId" = f2."followedUserId" AND f2."followerUserId" = $2
+  WHERE f1."followedUserId" = $1;
+`;
+
+const getFollowingUserList = `
+  SELECT
+    u."userId",
+    u."username",
+    u."displayName",
+    CASE WHEN f2."followerUserId" IS NOT NULL THEN TRUE ELSE FALSE END AS "isFollowing"
+  FROM app_user u
+  INNER JOIN follow f ON u."userId" = f."followedUserId"
+  LEFT JOIN follow f2 ON u."userId" = f2."followedUserId" AND f2."followerUserId" = $2
+  WHERE f."followerUserId" = $1;
+`;
```

Comment:
We don't need the self join here. We need it when looking for followers because for example if we're looking for user A's followers, we need to know if user A is following those people back. However, here user A would always be following their followers, so we can do something simpler. We also don't need 2 user id's:

```suggestion
const getFollowingUserList = `
  SELECT
    u."userId",
    u."username",
    u."displayName",
    TRUE AS "isFollowing"
  FROM app_user u
  INNER JOIN follow f ON u."userId" = f."followedUserId"
  WHERE f."followerUserId" = $1;
`;
```

Reply:
This makes more sense. No need to do that case stuff as it is always  true and no need to have a left join fr. Simpler and makes sense

Reply:
Wait, correct me if I am wrong. I was working on the assumption that the `isFollowing` status is based on the currently logged in user.

Like, you will fetch the users that the visited profile is following and the `isFollowing` is dependent on if the current user that is logged is following the users that have been fetched.

Thats why I have two userIds, one for the visited profile and another for the currentuser profile.

Based on my reasoning, if I implement your suggestion, it will always return true yet that isn't case.

Reply:
I think you're right, my bad. I missed that we need to determine the `isFollowing` status when someone it looking at another person's profile